### DISCUSSION TITLE
Ordinal in String.Equals in C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#144] [Q&A] [answered]
Hi,

I didn't understand how Ordinal exactly works. Could you please explain? How does Ordinal work in the background?

Microsft said:

> Compare strings using ordinal (binary) sort rules.

**Comment by christiannagel** [comment, marked as answer]
Sorting strings depends on the culture. The table in the following link gives the order for the value cote, coté, côte, côté, Namibia... with English-US, French-France, and Spanish-Spain:

[Sort order sample table](https://learn.microsoft.com/en-us/globalization/locale/sorting-and-string-comparison#how-do-i-verify-that-sorting-works-correctly-in-my-code?WT.mc_id=DT-MVP-10160)

Using different cultures, the sort order not only depends on special characters as shown in the previous table, but there are also differences how some ASCII characters are sorted. One language I came across had the same sort order for two different characters, thus here the next character was important how a string was sorted.

With ordinal sort rules, also *binary* was mentioned. This gives a hint. Here the sort is done based on the binary number value. 

For whom are you sorting strings? 

* For people: use the culture. 
* For the system, you can use the invariant culture (which has some rules independent of a culture), or ordinal sort which just uses the number value.
